NOpenType/Renderer.cs

```diff
@@ -179,24 +187,130 @@ private static Point<int> GetMidPoint(Point<int> v1, Point<int> v2)
+            var glyphs = Layout(0, 0, text, options);
+
+            int scalingFactor = ScalingFactor(options.FontSize);
+
+            var right = glyphs.Max(x => x.right);
+            var left = glyphs.Min(x => x.left);
+            var bottom = glyphs.Max(x => x.bottom);
+            var top = glyphs.Min(x => x.top);
+
+            return new Size()
+            {
+                Width = ((right - left) * scalingFactor) / Constants.FontToPixelDivisor,
+                Height = ((bottom - top) * scalingFactor) / Constants.FontToPixelDivisor
+            };
         }
 
-        public void Render(int x, int y, string text, int size, int resolution)
+        /// <summary>
+        /// Layouts the <paramref name="text"/> at point specified by <paramref name="x"/> and <paramref name="y"/>
+        /// based on the <paramref name="TextOptions"/>.
+        /// </summary>
+        /// <param name="x">The x.</param>
+        /// <param name="y">The y.</param>
+        /// <param name="text">The text.</param>
+        /// <param name="options">The options.</param>
+        /// <returns>Returns a collection of <see cref="GlyphLayout"/> that describe the layout positioning of each glyph.</returns>
+        private IEnumerable<GlyphLayout> Layout(int x, int y, string text, TextOptions options)
         {
-            int xx = x * EmSquare.Size * PointsPerInch;
-            int yy = y * EmSquare.Size * PointsPerInch;
-            int m = size * resolution;
-            int d = EmSquare.Size * PointsPerInch;
+            //we are working a font sizes in here
+            //convert from pixel sizes to font sizes
+            int scalingFactor = ScalingFactor(options.FontSize);
+            int xx = (int)(x * Constants.FontToPixelDivisor) / scalingFactor;
+            int yy = (int)(y * Constants.FontToPixelDivisor) / scalingFactor;
+
+            int drawheightEM = _typeface.Bounds.YMax - _typeface.Bounds.YMin;
+            int lineHeight = (int)(drawheightEM * options.LineHeight);
+
             foreach (var character in text)
             {
-                RenderGlyph(xx, yy, m, d, _typeface.Lookup(character));
-                xx += _typeface.GetAdvanceWidth(character);
+                var glyph = _typeface.Lookup(character);
+                var glyphWidth = _typeface.GetAdvanceWidth(character);
+                yield return new GlyphLayout
+                {
+                    glyph = glyph,
+                    bottom = lineHeight + yy,
+                    right = glyphWidth + xx,
+                    left = xx,
+                    top = yy
+                };
+
+                xx += glyphWidth;
             }
-            _rasterizer.Flush();
         }
-    }
 
+        internal struct GlyphLayout
+        {
+            public Glyph glyph;
+            public int left;
+            public int top;
+            public int bottom;
+            public int right;
+        }
+    }
```

Comment:
The bottom right is required for finding the right most extent of the bounding box and thus the width and the bottom is the same for the height.

If I don't have this struct we would need to ask the typeface multiple times for the glyph and the advance width etc.